text/0000-lang-team-advisors.md

```diff
@@ -0,0 +1,106 @@
+
+## Adding a new lang-team advisor
+
+Lang team advisors are added through the [same general process as regular lang team members][new-member-process]:
+
+[new-member-process]: https://github.com/rust-lang/lang-team/pull/174/
+
+* Any lang team member can send a message with a proposal for lang-team advisors.
+    * The message should include a short write-up giving answers to the questions below. It is particularly useful to provide examples (e.g., we sought their opinion at this point).
+
+These are the questions we ask ourselves when deciding whether someone would be a good choice as a lang team advisor.
+
+* Do we regularly seek this person's opinion when **deliberating**?
+    * For example, during triage, do we often say "let's check what this person thinks".
+* Does this person have particular knowledge of some domain, or some particular part of Rust? Alternatively, do they have broad knowledge of Rust?
+* If, even after a long and protected debate, you knew that this person had concerns about a design, would you want to block the design from going forward until you had a chance to hear them out?
+
+Naturally, the [questions for lang team membership][new-member-process] are also appropriate, but they are "nice to haves", the bar for an advisor is lower.
+
+When adding an advisor primarily for a specific area of expertise, we should document that area of expertise in a comment in the lang-team-advisors toml file.
+
+## Removing a lang-team advisor
+
+An advisor may be removed at their request, or if the team feels they've been inactive for an extended period. However, advisors (like any team member) are free to take vacations and otherwise maintain life/Rust balance.
+
+## Integration into the decision process
+
+There will be a team in the rust repo (`rust-lang/lang-team-advisors`). When a lang team FCP is initiated, we will cc this team, making them aware it is happening. Advisors will be able to raise blocking objections with the "concern" functonality of rfcbot, or equivalent functionality in future decision tooling. (As an interim measure until rfcbot includes this functionality, team members may raise concerns on behalf of advisors on request.)
+
+The precise details of how advisors fit into the lang team [decision making process](https://lang-team.rust-lang.org/decision_process/reference.html) are as follows:
+
+* Like lang-team members, advisors may raise a blocking concern on an FCP. The expectation is that the advisor will work with the implementors to resolve the concern to everyone's mutual satisfaction.
+* Unlike lang-team members, advisors cannot sustain a concern to prevent it from being overruled; only full lang-team members can opt to sustain a concern. A concern raised by an advisor may be overruled if "all but one" lang-team members agree that it has been adequately addressed (this rule ensures that there is no incentive for an advisor to "proxy" a concern on behalf of a full member).
```

Comment:
> * Unlike lang-team members, advisors cannot sustain a concern to prevent it from being overruled; only full lang-team members can opt to sustain a concern. A concern raised by an advisor may be overruled if "all but one" lang-team members agree that it has been adequately addressed (this rule ensures that there is no incentive for an advisor to "proxy" a concern on behalf of a full member).

I'm relatively unconcerned about that kind of "proxy" situation. I'm not sure if I want to *drop* this point entirely, but I will say I'm *tempted* to say a lang-team member *should* be able to "sustain" an objection by an advisor.

More importantly, though:
```suggestion
* Unlike lang-team members, advisors cannot sustain a concern to prevent it from being overruled; only full lang-team members can opt to sustain a concern. A concern raised by an advisor may be overruled if "all but one" lang-team members agree that it has been adequately heard and understood (this rule ensures that there is no incentive for an advisor to "proxy" a concern on behalf of a full member).
```

As with a lang-team member objection, the obligation isn't necessarily to "address" it; the obligation is to hear and fully understand it, and decide if there's something to address that hasn't been.